### DISCUSSION TITLE
Discussion on Tux's sliding ability

### DISCUSSION THREAD

**Discussion by Alasdairbugs** [SuperTux/supertux#2467] [General]
There's been a lot of discussion in the supertux discord about the sliding mechanics in the recent nightlies. I've compiled a list of current issues:

1. the sprite used for sliding is a still image, and tux's belly is not squished by the rest of his body weight (my personal opinion).

the next problem may be part of several problems;

2. you can chain a lot of momentum from "bunny-hop" style slide-jumping on flat surfaces. it can be argued that only slopes should be able to grant tux extra momentum (from max running speed)

now if somehow, tux is able to only explicitly slide off of slopes, i imagine that the "corner-boost" (a phenomenon where the game's corner correction system displaces tux a few frames out of a hitbox if slightly overlapping vertically) could still be used to give tux large slide momentum, which would also need to be fixed.

**Comment by Alasdairbugs** [comment]
i think 1. could be fixed with two things

- a particle animation of snow that increases in intensity as tux's sliding speed increases on the ground; alternative particles can be grass or dirt (forest), sand and etc. this is similar to the particles for splashing in and out of water planned.
- some resprite of tux's sliding that squishes his belly while sliding, and perhaps very slight rotation correlated with the slope angle.

i think 1. is definitely low priority and probably an 0.8+ feature, but possible.

**Reply by autofly17** [reply]
I think the current sliding mechanic is in itself lacking. If I recall, it was originally designed for it to be used to gain momentum and make large jumps on slopes.

![Slider2](https://user-images.githubusercontent.com/130930373/236960288-b193a945-710d-4eba-b78b-e73cac4ff917.png)

Currently it's just a quick slide that's always facing straight with no real advantage. I think if we added some special tiles for slopes like ice/mud that instantly make tux slide with his current momentum it would be a more fun mechanic.

However, I think all the specifics of this design make it too ambitious to be added anytime in the next release. We should just stick with debugging the current behavior and innovate on it later in 0.8+

**Reply by provigz** [reply]
If there's something to improve on sliding, in my opinion those fixes should be done for 0.7.0, not rush it.
